### ISSUE TITLE
Error on building .so for Android

### ISSUE THREAD

**Issue by sj120698** [microsoft/onnxruntime-extensions#153]
I am getting an error on building android .so for onnxruntime-extensions library using command - 

```
cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=${NDK_ROOT}/build/cmake/android.toolchain.cmake 
-DANDROID_NDK=${NDK_ROOT} -DANDROID_ABI=arm64-v8a -DANDROID_PLATFORM=android-19 
-DOCOS_ENABLE_STATIC_LIB=OFF ../../.. && cmake --build . --config Release --parallel

```
Error - 
```
[153/153] Linking CXX shared library libortcustomops.so
FAILED: libortcustomops.so
cmd.exe /C "cd . && C:\Users\shubham.j\AppData\Local\Android\Sdk\ndk\21.0.6113669\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe --target=aarch64-none-linux-android21 --gcc-toolchain=C:/Users/shubham.j/AppData/Local/Android/Sdk/ndk/21.0.6113669/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=C:/Users/shubham.j/AppData/Local/Android/Sdk/ndk/21.0.6113669/toolchains/llvm/prebuilt/windows-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security   -O2 -g -DNDEBUG  -Wl,--exclude-libs,libgcc_real.a -Wl,--exclude-libs,libatomic.a -static-libstdc++ -Wl,--build-id -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments -shared -Wl,-soname,libortcustomops.so -o libortcustomops.so CMakeFiles/ortcustomops.dir/shared/ortcustomops.cc.o  libocos_operators.a  _deps/googlere2-build/libre2.a  _deps/sentencepieceproject-build/src/libsentencepiece.a  -pthread  _deps/blingfire-build/libbingfirtinydll_static.a  _deps/blingfire-build/libfsaClientTiny.a  -latomic -lm && cd ."
_deps/sentencepieceproject-build/src/libsentencepiece.a(common.cc.o): In function `google::protobuf::internal::DefaultLogHandler(google::protobuf::LogLevel, char const*, int, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&)':
C:\Users\shubham.j\Onnx_KFA\onnxruntime-extensions\out\android\Release/_deps/sentencepieceproject-src/third_party/protobuf-lite/common.cc:150: undefined reference to `__android_log_write'
C:\Users\shubham.j\Onnx_KFA\onnxruntime-extensions\out\android\Release/_deps/sentencepieceproject-src/third_party/protobuf-lite/common.cc:158: undefined reference to `__android_log_write'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.

```
I am able to build static .a by setting DOCOS_ENABLE_STATIC_LIB=ON in the cmake command. But setting the flag to OFF, yields the above error. I require the .so file to register the library in ONNX runtime session with -

`Ort::ThrowOnError(Ort::GetApi().RegisterCustomOpsLibrary((OrtSessionOptions*)session_options, custom_op_library_filename, &handle));
`

**Comment by wenbingl** [comment]
Thanks for report this issue.
We haven't officially release Android version yet, which will happen after 0.4.0 release.

for this specific issue, it looks the sentencepiece library that the project depends on, it might already have some solution if you check their website.

**Comment by sj120698** [comment]
The error can be resolved in v0.4.0 by adding log-lib to - https://github.com/microsoft/onnxruntime-extensions/blob/4ebb8667da7136178b04496fe6d87c27199fe8e9/CMakeLists.txt#L333 

this can be done as following - 

```
find_library( # Sets the name of the path variable.
              log-lib
              # Specifies the name of the NDK library that
              # you want CMake to locate.
              log )

target_compile_definitions(ortcustomops PRIVATE  ${OCOS_COMPILE_DEFINITIONS})
target_link_libraries(ortcustomops PUBLIC ocos_operators ${log-lib})
```
